csrc/scheduler/registry_utils.cpp

```diff
@@ -442,63 +442,77 @@ bool reductionInterferingView(
-              (int64_t)disjoint_set_information.disjoint_set_ids.size(),
-          "Error computing disjoint group on the logical domain of ",
-          reduction_reference->toString());
-      disjoint_id_sets.push_back(
-          disjoint_set_information.disjoint_set_ids[logical_pos]);
+  // Check if any view op could merge IDs in different coalesced_ids.
+  // (1) Build the exact graph
+  IdModel id_model(fusion, false, false, false);
+  id_model.buildExactGraph();
+  const ValGraph& exact_graph = id_model.idGraph(IdMappingMode::EXACT);
+
+  // (2) For each ID in vect_of_coalesced_ids, find its corresponding ValGroup
+  // and save the mapping info. ValGroup is a set of equivalent IDs.
+  // Map through: ID --> ValGroup --> coalesced_ids (index in the vector)
+  std::unordered_map<ValGroup, int> vg_to_coalesced_ids;
+  for (auto index : c10::irange(vect_of_coalesced_ids.size())) {
+    for (auto id : vect_of_coalesced_ids.at(index)) {
+      const auto& vg = exact_graph.toGroup(id);
+      vg_to_coalesced_ids[vg] = (int)index;
     }
-    disjoint_groups.push_back(disjoint_id_sets);
   }
 
-  // Make sure there's no intersection between the groups, otherwise view
-  // will interfere with the schedule. TODO: Make this better complexity,
-  // since it should be relatively small int vectors of a small total nDims,
-  // not too worried about it now.
-
-  for (auto first_dim_i : c10::irange(disjoint_groups.size())) {
-    for (auto second_dim_i = first_dim_i + 1;
-         second_dim_i < disjoint_groups.size();
-         ++second_dim_i) {
-      auto first_group = disjoint_groups[first_dim_i];
-      auto second_group = disjoint_groups[second_dim_i];
-      for (auto first_disjoint_id : first_group) {
-        for (auto second_disjoint_id : second_group) {
-          if (first_disjoint_id == second_disjoint_id) {
-            return true;
-          }
+  // (3) Loop through all view ops and check if they merge IDs in different
+  // coalesced_ids. If happens, return true.
+  for (auto view : ir_utils::getViewOps(fusion)) {
+    auto tv = view->out();
+    // visit exprs between root and logical domains in topologically sorted
+    // order (can't use id_model.idUses() since it is not sorted).
+    // If the expr inputs exist in [vg_to_coalesced_ids], add the newly created
+    // IDs to the same coalesced_ids as the input IDs. Otherwise, do nothing
+    // since that expr is not interfering with IDs of reduction tv.
+    for (auto expr : StmtSort::getExprsTo(
+             {tv->getLogicalDomain().begin(), tv->getLogicalDomain().end()})) {
+      if (auto merge = dynamic_cast<Merge*>(expr)) {
+        const auto& vg0 = exact_graph.toGroup(merge->inner());
+        const auto& vg1 = exact_graph.toGroup(merge->outer());
+        auto it0 = vg_to_coalesced_ids.find(vg0);
+        auto it1 = vg_to_coalesced_ids.find(vg1);
+        if (it0 == vg_to_coalesced_ids.end() ||
+            it1 == vg_to_coalesced_ids.end()) {
+          continue;
```

Comment:
Is it ok to just ignore this case?

Reply:
This and the similar one for split only happens when the reshape is before reduction. If reshape is after reduction, its `root domain` is same to reduction tv's logical domain, so all the IDs constructed from root exists in `vg_to_coalesced_ids`.
If reshape is before reduction, we can actually skip all the checks since the reduction tv should have the same logical domain as the reshape output tv. 
I did the following changes:
```
(1) skip all view ops before reduction
(2) change from `continue` to `NVF_ERROR`
```